packages/select/src/vaadin-select.js

```diff
@@ -529,34 +574,71 @@ class Select extends DelegateFocusMixin(FieldMixin(SlotMixin(ElementMixin(Themab
 
     const label = selected.getAttribute('label');
     if (label) {
-      labelItem = this.__createItem(label);
+      labelItem = this.__createItemElement({ label });
     } else {
       labelItem = selected.cloneNode(true);
     }
 
     // store reference to the original item
     labelItem._sourceItem = selected;
 
-    this.__appendItem(labelItem);
+    this.__appendValueItemElement(labelItem);
 
     // ensure the item gets proper styles
     labelItem.selected = true;
   }
 
-  /** @private */
-  __createItem(text) {
-    const item = document.createElement('vaadin-item');
-    item.textContent = text;
-    return item;
+  /**
+   * @param {!SelectItem} item
+   * @private
+   */
+  __createItemElement(item) {
+    let itemElement;
+    if (item.component instanceof HTMLElement) {
+      itemElement = item.component;
+    } else {
+      itemElement = document.createElement(item.component || 'vaadin-item');
+    }
+
+    if (item.label) {
+      itemElement.textContent = item.label;
+    }
+    if (item.value) {
+      itemElement.value = item.value;
+    }
+    if (item.disabled) {
+      itemElement.disabled = item.disabled;
+    }
```

Comment:
Just want to point out that these might also be set on elements created from a custom component. Can not see how that would be harmful. Maybe that's intentional (custom item renderer component that supports same API as `vaadin-item`).

Reply:
Yeah, I did it intentionally in order to make it possible to get these properties added to a component with the same API as `vaadin-item`. Well, now I'm quite uncertain if it is actually required. @tomivirkki What do you think?

Reply:
I don't see a problem applying these properties regardless of the component. One could for example create a group title for some options with
```js
{
  component: 'b',
  label: 'Other'
}
```

I was thinking we could later consider `itemRenderer` API for further customization of the individual items

```js
// root would the result of document.createElement(item.component || 'vaadin-item')
select.itemRenderer = (root, itemModel) => {}
```

...where the current behaviour would be the "default renderer".

Reply:
Thanks for the feedback, resolving this comment for now.